src/Yahoo/Provider.php

```diff
@@ -57,11 +60,11 @@ protected function getUserByToken($token)
     protected function mapUserToObject(array $user)
     {
         return (new User())->setRaw($user)->map([
-            'id'       => $user['guid'],
+            'id'       => $user['sub'],
             'nickname' => $user['nickname'],
```

Comment:
What if the nickname is null?
I would fallback to sub

Reply:
ok, i will add it